pkg/pgclient/client_test.go

```diff
@@ -183,3 +184,57 @@ func TestHealthCheck(t *testing.T) {
 		t.Fatal("health check method not called when expected")
 	}
 }
+
+func TestNewClient(t *testing.T) {
+	type args struct {
+		cfg          *Config
+		schemaLocker LockFunc
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    *Client
+		wantErr bool
+		error   error
+	}{
+		{
+			"Test failure case when user provides both --db-uri & --db flags with db creds.",
+			args{
+				cfg: &Config{
+					Host:                    "localhost",
+					Port:                    5433,
+					User:                    "postgres",
+					Password:                "Timescale123",
+					Database:                "timescale",
+					SslMode:                 "require",
+					DbConnectRetries:        0,
+					AsyncAcks:               false,
+					ReportInterval:          0,
+					LabelsCacheSize:         0,
+					MetricsCacheSize:        0,
+					SeriesCacheSize:         0,
+					WriteConnectionsPerProc: 1,
+					MaxConnections:          0,
+					UsesHA:                  false,
+					DbUri:                   "postgres://postgres:password@localhost:5432/postgres?sslmode=allow",
+				},
+				schemaLocker: nil,
+			},
+			nil,
+			true,
+			errors.New("failed to build DB credentials with provided flags. Please use either db flags or db-uri not both"),
```

Comment:
If we are testing for this exact error, it should be made into a variable and reused in the test. Also, no need to use `errors` package, `fmt.Errorf` is used everywhere to make new errors and its already imported.